lib/cosmos/utilities/low_fragmentation_array.rb

```diff
@@ -8,4 +8,12 @@
 # as published by the Free Software Foundation; version 3 with
 # attribution addendums as found in the LICENSE.txt
 
-require 'cosmos/ext/low_fragmentation_array'
+if RUBY_ENGINE == 'ruby' and !ENV['COSMOS_NO_EXT']
+  require 'cosmos/ext/low_fragmentation_array'
+else
+  module Cosmos
+    class LowFragmentationArray < Array
```

Comment:
It'll be interesting to benchmark Jruby since it's a memory hog already (from what I understand).